lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -1099,6 +1098,14 @@ void request_disconnect(DAP &dap, const llvm::json::Object &request) {
     dap.broadcaster.BroadcastEventByType(eBroadcastBitStopProgressThread);
     dap.progress_event_thread.join();
   }
+  if (dap.stdout_forward_thread.joinable()) {
+    dap.pout.Close();
```

Comment:
This may be moot due to due other issue, but you should only close the write end before joining the thread (the read end can be closed after joining, or from within the thread). Closing the read descriptor while the thread is running leads to a race with the `Read` call inside the thread.

Reply:
Done.